src/main/java/frc/team166/robot/subsystems/Lift.java

```diff
@@ -324,6 +327,31 @@ protected boolean isFinished() {
         };
     }
 
+    public Command GoToTopLift() {
+        return new SubsystemCommand(this) {
+            @Override
+            protected void initialize() {
+                disengageBrake();
+                ShiftToHighGear();
+            }
+
+            @Override
+            protected void execute() {
+                if (topLimitSwitch.get() == true) {
```

Comment:
This if should not exist here. Instead the isFinished function below should be returning `!topLimitSwitch.get()` and there should be an end function that stops the drive motor and engages the brake

Reply:
Ben is right, and I think this will apply to the "GoToBottomLift" command as well... The Execute's for these should just be driving the lift up or down (positive or negative speeds). The IsFinished's should just be looking for either the top or bottom limit switch to be set. I can't remember if that digital input will return a true or a false when the switch is pressed, but you should be able to check that with SmartDash. Then in the IsFinished's all you need is `return topLimitSwitch.get()` or `return !topLimitSwitch.get()`